web/assets/js/util/utils.js

```diff
@@ -146,19 +143,15 @@ class RandomUtil {
 
     static randomShadowsocksPassword() {
         let array = new Uint8Array(32);
-        window.crypto.getRandomValues(array);
+        crypto.getRandomValues(array);
         return btoa(String.fromCharCode.apply(null, array));
     }
 
     static randomShortId() {
-        let shortIds = ['','','',''];
-        for (var ii = 0; ii < 4; ii++) {
-            for (var jj = 0; jj < this.randomInt(8); jj++){
-                let randomNum = this.randomInt(256);
-                shortIds[ii] += ('0' + randomNum.toString(16)).slice(-2)
-            }
-        }
-        return shortIds;
+        const randomBytes = new Uint8Array(4);
+        crypto.getRandomValues(randomBytes);
+        const shortId = Array.from(randomBytes).map(byte => byte.toString(16).padStart(2, '0')).join('');
+        return shortId;
```

Comment:
1. This code will create only one `shortId` with static lenght which is less secure than variable length. The string's length should be between 0-16 but multiple of 2.
2. This code will cause type problem due to expectation is an array. Previous array lenght was 4.

Please change it to return an array of separate `shortId`s with random length.

Reply:
Right, sure

Reply:
The question that is raised is why the shortIds returns as a array, but only [0] is used?
Isn't it better to return only one string instead of several indexes and place it in shortIds[0] ?

Reply:
Using array of shortIds is based on xray-core. If someone needs another shortId to change client config manually, it should be possible.

Regarding using only first shortId in qrcode and config providers, it is about missing unique idea for assiging and creating shortId.
The best idea was adding one shortId to the `inbound` when new client added. But it may cause several issues in large amount of clients.
Another idea was to define 8 or 4 initial values and assign them based on client index cycle.
And other solutions which may have several pros and cons.

Till finding the best solution, config providers will use first index of shortId.

Reply:
OK, i got it your mean, i will apply your said on shortIds functions and pull request it again. But about randomUUID, are you agree generate that with crypto.getRandomValue ?

Reply:
Hello, please return the panel to the previous state. There are problems in the panel of the Aziraza version, such as the number of users created from one number to another. It is difficult to create a user from 300 users and above. It was because we were using your panel in the Alireza panel. When the number of users increases, the panel slows down and creating users is difficult. In version 3, we used to create 1000 users on each panel and there was no problem. The panel did not slow down, but now it slows down.